pyscript.core/docs/README.md

```diff
@@ -0,0 +1,481 @@
+
+```html
+<script type="module">
+    import {
+        define,      // define a custom type="..."
+        whenDefined, // wait for a custom type to be defined
+        XWorker      // allows JS <-> Interpreter communication
+    } from 'https://esm.run/@pyscript/core';
+</script>
+```
+
+Please keep reading this document to understand how to use those utilities or how to have other *Pogramming Languages* enabled in your page via `<script>` elements.
+
+
+## How Scripts Work
+
+The [&lt;script&gt; element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script) has at least these extremely important peculiarities compared to any other element defined by the [HTML Standard](https://html.spec.whatwg.org/multipage/):
+
+ * its only purpose is to contain *data blocks*, meaning that browsers will never try to parse its content as generic *HTML* (and browsers will completely ignore either its content or its attributes, including the `src`, when its [type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type) is not known)
+ * its completely unobtrusive when it comes to both *aria* and *layout*, indeed it's one of the few nodes that can be declared almost anywhere without breaking its parent tree (other notable exception would be a comment node)
+ * for our specific use case, it already offers attributes that are historically well understood and known, also simplifying somehow the creation of this document
+
+The long story short is that any `<script type="non-standard-type">` has zero issues with any browser of choice, but it's true that using some specific *custom type* might lead to future issues in case that `type` could have some special meaning for the future of the Web.
+
+We encourage everyone to be careful when using this *core* API as we definitively don't want to clash or conflict, by any mean, with what the Web might need or offer in the near to far future, but we're also confident so far our current *types* are more than safe.
+
+### Script Attributes
+
+| name      | example                                       | behavior |
+| :-------- | :-------------------------------------------- | :--------|
+| async     | `<script type="pyodide" async>`               | The code is evaluated via `runAsync` utility where, if the *interpreter* allows it, top level *await* would be possible, among other *PL* specific asynchronous features.  |
+| config    | `<script type="pyodide" config="./cfg.toml">` | The interpreter will load and parse the *JSON* or *TOML* file to configure itself. Please see [currently supported config values](https://docs.pyscript.net/latest/reference/elements/py-config.html#supported-configuration-values) as this is currently based on `<py-config>` features. |
+| env       | `<script type="pyodide" env="brand">`         | Create, if not known yet, a dedicated *environment* for the specified `type`. Please read the [Terminology](#terminology) **env** dedicated details to know more. |
```

Comment:
Create, if not known yet, a dedicated unique *environment* for the specified `type`. If that environment has already been created, it'll reuse it instead. Please read the [Terminology](#terminology) **env** dedicated details to know more.